modules/api-compatibility-guidelines.adoc

```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+//
+// * rest_api/understanding-compatibility-guidelines.adoc
+
+[id="api-compatibility-guidelines_{context}"]
+= API compatibility guidelines
+
+Red Hat recommends that application developers adopt the following principles in order to improve compatibility with {product-title}.
+
+* Use APIs and components with support tiers that match the application's need.
+* Build applications using the published client libraries where possible.
+* Applications are only guaranteed to run correctly if they execute in an environment that is as new as the environment it was built to execute against.  For example, an application that was built for {product-title} 4.6 is not guaranteed to function properly on {product-title} 4.7.
```

Comment:
I think this should read:

"An application that was built for 4.7 is not guaranteed to function properly on 4.6"